compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPULayoutAnalysisAndDistribution.cpp

```diff
@@ -97,35 +97,37 @@ enum class MMAMatrixType { AMatrix, BMatrix, CMatrix };
 
 static std::array<Dimension, 3>
 getMMADimensions(MMAType mmaType, MMAMatrixType matrixType, int dim) {
-  switch (mmaType) {
-  case MMAType::M16N8K16:
-    switch (matrixType) {
-    case MMAMatrixType::AMatrix:
-      if (dim == 0)
-        return {{{DimType::LaneIdY, 8},
-                 {DimType::VecIdZ, 2},
-                 {DimType::LaneIdZ, 1}}};
+  if (mmaType != MMAType::M16N8K16) {
+    return {};
+  }
+
+  switch (matrixType) {
+  case MMAMatrixType::AMatrix: {
+    if (dim == 0) {
       return {
-          {{DimType::VecIdX, 2}, {DimType::LaneIdX, 4}, {DimType::VecIdY, 2}}};
-    case MMAMatrixType::BMatrix:
-      if (dim == 0)
-        return {{{DimType::LaneIdY, 8},
-                 {DimType::LaneIdZ, 1},
-                 {DimType::VecIdZ, 1}}};
+          {{DimType::LaneIdY, 8}, {DimType::VecIdZ, 2}, {DimType::LaneIdZ, 1}}};
+    }
+    return {
+        {{DimType::VecIdX, 2}, {DimType::LaneIdX, 4}, {DimType::VecIdY, 2}}};
```

Comment:
I also considered creating a local function so that we don't have to fight the nesting of braces in the first place, e.g.:
```c++
static std::array<Dimension, 3> createDims(Dimension a, Dimension b, Dimension c) { ... }
...

    case MMAMatrixType::CMatrix:
      if (dim == 0)
        return createDims({DimType::LaneIdY, 8}, {DimType::VecIdY, 2},
                          {DimType::LaneIdZ, 1});
```

WDYT?

Reply:
I don't think that's worth adding a new function for, and that looks about the same to me.